packages/format-library/src/link/inline.js

```diff
@@ -148,14 +151,37 @@ function InlineLinkUI( {
 					newText.length
 				);
 
```

Comment:
Commenting on folded code above here. What happens to _existing_ formats? I see that you're creating a new rich text value with the new text, and then applying a link format to it. But by replacing the exiting selection with this new rich text value, any existing formatting will be removed (e.g. bold or italic).

Perhaps this is the wrong way to approach the text replacement? I think we should do the replacement in two steps:
1) Replace the link format's URL as we would normally do.
2) Replace the active link's text only.

Reply:
Yeh clobbering other formats doesn't sound great 😰